docs/topics/addons.rst

```diff
@@ -0,0 +1,348 @@
+.. _topics-addons:
+
+=======
+Add-ons
+=======
+
+Scrapy's add-on system is a framework which unifies managing and configuring
+components that extend Scrapy's core functionality, such as middlewares,
+extensions, or pipelines. It provides users with a plug-and-play experience in
+Scrapy extension management, and grants extensive configuration control to
+developers.
+
+
+Activating and configuring add-ons
+==================================
+
+Add-ons and their configuration live in Scrapy's
+:class:`~scrapy.addons.AddonManager`. During Scrapy's start-up process, and
+only then, the add-on manager will read a list of enabled add-ons and their
+configurations from your ``ADDONS`` setting.
+
+The ``ADDONS`` setting is a dict in which every key is an addon class or its
+import path and the vaoue is its priority.
+
+The configuration of an add-on, if necessary at all, is stored as a dictionary
+setting whose name is the uppercase add-on name.
+
+This is an example where two add-ons (in this case with one requiring no
+configuration) are enabled/configured in a project's ``settings.py``::
+
+    ADDONS = {
+        'path.to.someaddon': 0,
+        path.to.someaddon2: 1,
+    }
+
+    SOMEADDON = {
+        'some_config': True,
+    }
+
+Enabling and configuring add-ons within Python code
+---------------------------------------------------
+
+The :class:`~scrapy.addons.AddonManager` will only read from Scrapy's settings
+*at the beginning* of Scrapy's start-up process.
```

Comment:
This is not precise; it's not about Scrapy start-up, it's about spider/crawler startup.